kolibri/auth/test/test_collections_and_roles.py

```diff
@@ -115,34 +122,89 @@ class CollectionsTestCase(TestCase):
+        self.assertEqual(Role.objects.filter(user=user, kind=Role.KIND_COACH, collection=self.facility).count(), 1)
+        self.classroom.remove_coach(user)
+        self.facility.remove_coach(user)
+        self.assertEqual(Role.objects.filter(user=user, kind=Role.KIND_COACH, collection=self.classroom).count(), 0)
+        self.assertEqual(Role.objects.filter(user=user, kind=Role.KIND_COACH, collection=self.facility).count(), 0)
+
+    def test_add_coaches(self):
+        user1 = FacilityUser.objects.create(username='foo1', facility=self.facility)
+        user2 = FacilityUser.objects.create(username='foo2', facility=self.facility)
+        self.classroom.add_coaches([user1, user2])
+        self.facility.add_coaches([user1, user2])
+        self.assertEqual(Role.objects.filter(kind=Role.KIND_COACH, collection=self.classroom).count(), 2)
+        self.assertEqual(Role.objects.filter(kind=Role.KIND_COACH, collection=self.facility).count(), 2)
+
+    def test_add_admins(self):
+        user1 = FacilityUser.objects.create(username='foo1', facility=self.facility)
+        user2 = FacilityUser.objects.create(username='foo2', facility=self.facility)
+        self.classroom.add_admins([user1, user2])
+        self.facility.add_admins([user1, user2])
+        self.assertEqual(Role.objects.filter(kind=Role.KIND_ADMIN, collection=self.classroom).count(), 2)
+        self.assertEqual(Role.objects.filter(kind=Role.KIND_ADMIN, collection=self.facility).count(), 2)
+
+    def test_add_classroom(self):
         classroom = Classroom.objects.create(parent=self.facility)
-        classroom.add_coach(user)
-        self.assertEqual(Role.objects.filter(user=user, kind=Role.KIND_COACH, collection=classroom).count(), 1)
+        self.assertEqual(Classroom.objects.count(), 2)
+        self.assertEqual(classroom.get_facility(), self.facility)
 
     def test_add_learner_group(self):
-        classroom = Classroom.objects.create(parent=self.facility)
+        classroom = Classroom.objects.create(name="blah", parent=self.facility)
+        classroom.full_clean()
         LearnerGroup.objects.create(parent=classroom)
         self.assertEqual(LearnerGroup.objects.count(), 1)
 
     def test_learner(self):
         user = FacilityUser.objects.create(username='foo', facility=self.facility)
         classroom = Classroom.objects.create(parent=self.facility)
-        learner_group = LearnerGroup.objects.create(parent=classroom)
+        learner_group = LearnerGroup.objects.create(name="blah", parent=classroom)
+        learner_group.full_clean()
         learner_group.add_learner(user)
         self.assertEqual(Role.objects.filter(user=user, kind=Role.KIND_LEARNER, collection=learner_group).count(), 1)
 
+    def test_parentless_classroom(self):
+        classroom = Classroom(name="myclass")
+        # shouldn't be valid, because no parent was specified, and Classrooms can't be the root of the collection tree
```

Comment:
I'll triple-quote you on that!

At least with `kolibri manage test`, it doesn't seem to be printing it for me, so I guess we're OK. Maybe it only prints it when there's an error in the test, for more context.